src/components/UserForm/index.jsx

```diff
@@ -3,15 +3,20 @@ import React, { useEffect } from 'react';
 import { useDispatch, useSelector, shallowEqual } from 'react-redux';
 import PropTypes from 'prop-types';
 import { Link } from 'react-router-dom';
+import { useForm, Controller } from 'react-hook-form';
+import classNames from 'classnames';
+import { yupResolver } from '@hookform/resolvers';
+import * as yup from 'yup';
 
 import paths from 'pages/Router/paths';
 import { usersCleanUp } from 'state/actions/users';
-import { useChangeHandler, useFormatDate, useFormatMessage } from 'hooks';
-import { validateEmail } from 'utils';
-import './UserForm.scss';
+import { useFormatDate, useFormatMessage } from 'hooks';
 import DatePicker from '../DatePicker';
+import errorMessage from '../ErrorMessage';
+
+import './UserForm.scss';
 
-const UserForm = ({ isEditing, isProfile, user, setUser, action }) => {
+const UserForm = ({ isEditing, isProfile, user, action }) => {
```

Comment:
I'm not a big fan of `UserForm` having its own state and dispatching action. This component should be only a representational component and the state should be managed by other components like `User`.